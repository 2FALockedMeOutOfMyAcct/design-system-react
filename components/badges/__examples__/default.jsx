import React from 'react';
import Badges from '~/components/badges';
import IconSettings from '~/components/icon-settings';

class Example extends React.Component {
	static displayName = 'badgeExample';

	render() {
		return (
			<IconSettings iconPath="/assets/icons">
				<Badges/>
			</IconSettings>
		)
	}
}
