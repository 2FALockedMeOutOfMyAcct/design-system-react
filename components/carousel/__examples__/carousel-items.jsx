/* eslint-disable  import/prefer-default-export */
import React from 'react';
import Carousel from '~/components/carousel';
import CarouselItem from '~/components/carousel/private/carousel-item';

export const ITEMS = [
	{
		id: 1,
		heading: 'Carousel Item 1',
		description: 'Description for #1',
		src: 'assets/images/carousel/carousel-01.jpg',
		CTALabel: 'Get Started',
		href: 'https://www.salesforce.com',
	},
	{
		id: 2,
		heading: 'Carousel Item 2',
		description:
			"Description for #2 will be  much longer than others. It'll look ok",
		src: 'assets/images/carousel/carousel-02.jpg',
		CTALabel: 'Get Started',
		href: 'https://www.salesforce.com',
	},
	{
		id: 3,
		heading: 'Carousel Item 3',
		description: 'Description for #3',
		src: 'assets/images/carousel/carousel-03.jpg',
		CTALabel: 'Get Started',
		href: 'https://www.salesforce.com',
	},
	{
		id: 4,
		heading: 'Carousel Item 4',
		description: 'Description for #4',
		src: 'assets/images/carousel/carousel-02.jpg',
		CTALabel: 'Get Started',
		href: 'https://www.salesforce.com',
	},
	{
		id: 5,
		heading: 'Carousel Item 5',
		description: 'Description for #3',
		src: 'assets/images/carousel/carousel-01.jpg',
		CTALabel: 'Learn More',
		href: 'https://www.salesforce.com',
	},
	{
		id: 6,
		heading: 'Carousel Item 6',
		description: 'Description for #6',
		src: 'assets/images/carousel/carousel-03.jpg',
		CTALabel: 'Learn More',
		href: 'https://www.salesforce.com',
	},
	{
		id: 7,
		heading: 'Carousel Item 7',
		description: 'Description for #6',
		src: 'assets/images/carousel/carousel-02.jpg',
		CTALabel: 'Learn More',
		href: 'https://www.salesforce.com',
	},
];
