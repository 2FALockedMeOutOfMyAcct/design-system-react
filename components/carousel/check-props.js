/* Copyright (c) 2015-present, salesforce.com, inc. All rights reserved */
/* Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license */
/* eslint-disable import/no-mutable-exports */
/* eslint-disable max-len */

import deprecatedEventParameter from '../../utilities/warning/deprecated-event-parameter';
import getComponentDocFn from '../../utilities/get-component-doc';
import onlyOneOfProperties from '../../utilities/warning/only-one-of-properties';

let checkProps = function() {};

if (process.env.NODE_ENV !== 'production') {
	checkProps = function(COMPONENT, props, jsonDoc) {
		// Stub
	};
}

export default checkProps;
