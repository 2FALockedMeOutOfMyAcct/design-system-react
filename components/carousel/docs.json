{
    "component": "carousel",
    "status": "prod",
    "display-name": "Carousels",
    "SLDS-component-path": "/components/carousel",
    "url-slug": "carousels"
}
