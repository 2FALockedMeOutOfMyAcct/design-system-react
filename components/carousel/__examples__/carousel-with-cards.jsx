import React from 'react';
// `~` is replaced with design-system-react at runtime
import Card from '~/components/card';
import Carousel from '~/components/carousel';
import CarouselItem from '~/components/carousel/private/item';
import { carouselItems } from './carousel-items';

class CarouselWithCards extends React.Component {
	static displayName = 'CarouselExample - with card items';

	render() {
		const itemsComponents = carouselItems.map((item, index) => (
			<Card key={`Item{$index}`}>{item}</Card>
		));

		return (
			<div style={{ width: '100%', maxWidth: '100%' }}>
				<Carousel
					id="ExampleCarouselWithCardItems"
					itemsPerPanel={this.props.itemsPerPanel}
					items={itemsComponents}
				/>
			</div>
		);
	}
}

CarouselWithCards.defaultProps = {
	itemsPerPanel: 3,
};

export default CarouselWithCards;
