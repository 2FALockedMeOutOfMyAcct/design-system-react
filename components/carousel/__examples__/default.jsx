import React from 'react';
import Carousel from '~/components/carousel';
import CarouselItem from '~/components/carousel/private/item';
import { carouselItems } from './carousel-items';

class Example extends React.Component {
	static displayName = 'CarouselExample - 3 items';

	render() {
		return (
			<div style={{ width: '100%', maxWidth: '100%' }}>
				<Carousel
					id={`ExampleCarouselWith${this.props.itemsPerPanel}items`}
					itemsPerPanel={this.props.itemsPerPanel}
					items={carouselItems}
					hasPreviousNextPanelNavigation={this.props.hasNavigation}
				/>
			</div>
		);
	}
}

Example.defaultProps = {
	itemsPerPanel: 3,
	hasNavigation: false,
};

export default Example;
