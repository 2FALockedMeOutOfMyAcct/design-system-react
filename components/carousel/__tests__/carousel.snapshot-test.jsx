/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';

import Card from '~/components/card';
import Carousel from '~/components/carousel';
import { ITEMS } from '~/components/carousel/__examples__/carousel-items';

import { renderMarkup } from '../../../tests/snapshot-helpers';

import Default from '../__examples__/default';
import '../../../styles/carousel/carousel.css';

test('Carousel Component Default Snapshot - 1 item, no navigation', () => {
	expect(renderMarkup(Default)).toMatchSnapshot();
});
test('Carousel Component Default Snapshot - 3 items and navigation', () => {
	expect(
		renderMarkup(Default, {
			itemsPerPanel: 3,
			hasPreviousNextPanelNavigation: true,
		})
	).toMatchSnapshot();
});

test('Carousel Component With Custom Rendering', () => {
	const onRenderItem = (item) => (
		<div className="slds-carousel-custom-rendering">
			<div className="slds-carousel__content-title">
				{item.heading} With Custom Rendering
			</div>
			<div className="slds-carousel__image">
				<img src={item.src} alt={item.imageAssistiveText} />
			</div>
		</div>
	);

	expect(
		renderMarkup(Carousel, {
			id: 'Custom Rendering',
			itemsPerPanel: 2,
			items: ITEMS,
			onRenderItem,
		})
	).toMatchSnapshot();
});
