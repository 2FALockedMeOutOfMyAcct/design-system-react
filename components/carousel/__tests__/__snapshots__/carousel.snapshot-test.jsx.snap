// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel Component Default Snapshot - 1 tem, no navigation 1`] = `
"<div style=\\"max-width:1280px\\">
  <div class=\\"slds-carousel\\" id=\\"ExampleCarouselWithundefineditems\\">
    <div class=\\"slds-grid_vertical slds-col slds-path__scroller\\">
      <div class=\\"slds-grid slds-grid_align-center\\">
        <div class=\\"slds-carousel__stage slds-align_absolute-center\\">
          <div class=\\"slds-carousel__panels slds-is-relative\\" style=\\"transform:translateX(-1000000px)\\">
            <div id=\\"content-id-1\\" class=\\"slds-carousel__panel slds-m-horizontal_xx-small slds-list_horizontal\\" role=\\"tabpanel\\" aria-hidden=\\"false\\" aria-labelledby=\\"indicator-id-1\\" style=\\"max-width:NaNpx\\">
              <a href=\\"javascript:void(0);\\" class=\\"slds-carousel__panel-action slds-text-link_reset\\">
                <div class=\\"slds-carousel__image\\"><img src=\\"assets/images/carousel/carousel-01.jpg\\" alt=\\"Card image description\\" /></div>
                <div class=\\"slds-carousel__content\\">
                  <h2 class=\\"slds-carousel__content-title\\">Carousel Item 1</h2>
                  <p class=\\"slds-p-bottom_x-small slds-text-body_small\\" style=\\"min-height:40px\\">Description for #1</p><a class=\\"slds-button slds-button_neutral slds-button_outline-brand\\" href=\\"https://www.salesforce.com\\" target=\\"_blank\\">Get Started</a></div>
              </a>
            </div>
            <div id=\\"content-id-2\\" class=\\"slds-carousel__panel slds-m-horizontal_xx-small slds-list_horizontal\\" role=\\"tabpanel\\" aria-hidden=\\"false\\" aria-labelledby=\\"indicator-id-2\\" style=\\"max-width:NaNpx\\">
              <a href=\\"javascript:void(0);\\" class=\\"slds-carousel__panel-action slds-text-link_reset\\">
                <div class=\\"slds-carousel__image\\"><img src=\\"assets/images/carousel/carousel-02.jpg\\" alt=\\"Card image description\\" /></div>
                <div class=\\"slds-carousel__content\\">
                  <h2 class=\\"slds-carousel__content-title\\">Carousel Item 2</h2>
                  <p class=\\"slds-p-bottom_x-small slds-text-body_small\\" style=\\"min-height:40px\\">Description for #2 will be much longer than others. It&#x27;ll look ok</p><a class=\\"slds-button slds-button_neutral slds-button_outline-brand\\" href=\\"https://www.salesforce.com\\" target=\\"_blank\\">Get Started</a></div>
              </a>
            </div>
            <div id=\\"content-id-3\\" class=\\"slds-carousel__panel slds-m-horizontal_xx-small slds-list_horizontal\\" role=\\"tabpanel\\" aria-hidden=\\"false\\" aria-labelledby=\\"indicator-id-3\\" style=\\"max-width:NaNpx\\">
              <a href=\\"javascript:void(0);\\" class=\\"slds-carousel__panel-action slds-text-link_reset\\">
                <div class=\\"slds-carousel__image\\"><img src=\\"assets/images/carousel/carousel-03.jpg\\" alt=\\"Card image description\\" /></div>
                <div class=\\"slds-carousel__content\\">
                  <h2 class=\\"slds-carousel__content-title\\">Carousel Item 3</h2>
                  <p class=\\"slds-p-bottom_x-small slds-text-body_small\\" style=\\"min-height:40px\\">Description for #3</p><a class=\\"slds-button slds-button_neutral slds-button_outline-brand\\" href=\\"https://www.salesforce.com\\" target=\\"_blank\\">Get Started</a></div>
              </a>
            </div>
            <div id=\\"content-id-4\\" class=\\"slds-carousel__panel slds-m-horizontal_xx-small slds-list_horizontal\\" role=\\"tabpanel\\" aria-hidden=\\"false\\" aria-labelledby=\\"indicator-id-4\\" style=\\"max-width:NaNpx\\">
              <a href=\\"javascript:void(0);\\" class=\\"slds-carousel__panel-action slds-text-link_reset\\">
                <div class=\\"slds-carousel__image\\"><img src=\\"assets/images/carousel/carousel-02.jpg\\" alt=\\"Card image description\\" /></div>
                <div class=\\"slds-carousel__content\\">
                  <h2 class=\\"slds-carousel__content-title\\">Carousel Item 4</h2>
                  <p class=\\"slds-p-bottom_x-small slds-text-body_small\\" style=\\"min-height:40px\\">Description for #4</p><a class=\\"slds-button slds-button_neutral slds-button_outline-brand\\" href=\\"https://www.salesforce.com\\" target=\\"_blank\\">Get Started</a></div>
              </a>
            </div>
            <div id=\\"content-id-5\\" class=\\"slds-carousel__panel slds-m-horizontal_xx-small slds-list_horizontal\\" role=\\"tabpanel\\" aria-hidden=\\"false\\" aria-labelledby=\\"indicator-id-5\\" style=\\"max-width:NaNpx\\">
              <a href=\\"javascript:void(0);\\" class=\\"slds-carousel__panel-action slds-text-link_reset\\">
                <div class=\\"slds-carousel__image\\"><img src=\\"assets/images/carousel/carousel-01.jpg\\" alt=\\"Card image description\\" /></div>
                <div class=\\"slds-carousel__content\\">
                  <h2 class=\\"slds-carousel__content-title\\">Carousel Item 5</h2>
                  <p class=\\"slds-p-bottom_x-small slds-text-body_small\\" style=\\"min-height:40px\\">Description for #3</p><a class=\\"slds-button slds-button_neutral slds-button_outline-brand\\" href=\\"https://www.salesforce.com\\" target=\\"_blank\\">Learn More</a></div>
              </a>
            </div>
            <div id=\\"content-id-6\\" class=\\"slds-carousel__panel slds-m-horizontal_xx-small slds-list_horizontal\\" role=\\"tabpanel\\" aria-hidden=\\"false\\" aria-labelledby=\\"indicator-id-6\\" style=\\"max-width:NaNpx\\">
              <a href=\\"javascript:void(0);\\" class=\\"slds-carousel__panel-action slds-text-link_reset\\">
                <div class=\\"slds-carousel__image\\"><img src=\\"assets/images/carousel/carousel-03.jpg\\" alt=\\"Card image description\\" /></div>
                <div class=\\"slds-carousel__content\\">
                  <h2 class=\\"slds-carousel__content-title\\">Carousel Item 6</h2>
                  <p class=\\"slds-p-bottom_x-small slds-text-body_small\\" style=\\"min-height:40px\\">Description for #6</p><a class=\\"slds-button slds-button_neutral slds-button_outline-brand\\" href=\\"https://www.salesforce.com\\" target=\\"_blank\\">Learn More</a></div>
              </a>
            </div>
            <div id=\\"content-id-7\\" class=\\"slds-carousel__panel slds-m-horizontal_xx-small slds-list_horizontal\\" role=\\"tabpanel\\" aria-hidden=\\"false\\" aria-labelledby=\\"indicator-id-7\\" style=\\"max-width:NaNpx\\">
              <a href=\\"javascript:void(0);\\" class=\\"slds-carousel__panel-action slds-text-link_reset\\">
                <div class=\\"slds-carousel__image\\"><img src=\\"assets/images/carousel/carousel-02.jpg\\" alt=\\"Card image description\\" /></div>
                <div class=\\"slds-carousel__content\\">
                  <h2 class=\\"slds-carousel__content-title\\">Carousel Item 7</h2>
                  <p class=\\"slds-p-bottom_x-small slds-text-body_small\\" style=\\"min-height:40px\\">Description for #6</p><a class=\\"slds-button slds-button_neutral slds-button_outline-brand\\" href=\\"https://www.salesforce.com\\" target=\\"_blank\\">Learn More</a></div>
              </a>
            </div>
          </div>
        </div>
      </div>
      <ul class=\\"slds-carousel__indicators slds-col slds-text-align_center\\" role=\\"tablist\\">
        <li class=\\"slds-carousel__indicator slds-m-horizontal_xx-small\\" role=\\"presentation\\"><a id=\\"indicator-id-1\\" class=\\"slds-carousel__indicator-action slds-is-active\\" role=\\"tab\\" tabindex=\\"0\\" aria-selected=\\"true\\" aria-controls=\\"panel-1\\" title=\\"Visit Panel 1\\"><span class=\\"slds-assistive-text\\">Panel 1</span></a></li>
        <li class=\\"slds-carousel__indicator slds-m-horizontal_xx-small\\"
          role=\\"presentation\\"><a id=\\"indicator-id-2\\" class=\\"slds-carousel__indicator-action\\" role=\\"tab\\" tabindex=\\"-1\\" aria-selected=\\"false\\" aria-controls=\\"panel-2\\" title=\\"Visit Panel 2\\"><span class=\\"slds-assistive-text\\">Panel 2</span></a></li>
        <li class=\\"slds-carousel__indicator slds-m-horizontal_xx-small\\"
          role=\\"presentation\\"><a id=\\"indicator-id-3\\" class=\\"slds-carousel__indicator-action\\" role=\\"tab\\" tabindex=\\"-1\\" aria-selected=\\"false\\" aria-controls=\\"panel-3\\" title=\\"Visit Panel 3\\"><span class=\\"slds-assistive-text\\">Panel 3</span></a></li>
        <li class=\\"slds-carousel__indicator slds-m-horizontal_xx-small\\"
          role=\\"presentation\\"><a id=\\"indicator-id-4\\" class=\\"slds-carousel__indicator-action\\" role=\\"tab\\" tabindex=\\"-1\\" aria-selected=\\"false\\" aria-controls=\\"panel-4\\" title=\\"Visit Panel 4\\"><span class=\\"slds-assistive-text\\">Panel 4</span></a></li>
        <li class=\\"slds-carousel__indicator slds-m-horizontal_xx-small\\"
          role=\\"presentation\\"><a id=\\"indicator-id-5\\" class=\\"slds-carousel__indicator-action\\" role=\\"tab\\" tabindex=\\"-1\\" aria-selected=\\"false\\" aria-controls=\\"panel-5\\" title=\\"Visit Panel 5\\"><span class=\\"slds-assistive-text\\">Panel 5</span></a></li>
        <li class=\\"slds-carousel__indicator slds-m-horizontal_xx-small\\"
          role=\\"presentation\\"><a id=\\"indicator-id-6\\" class=\\"slds-carousel__indicator-action\\" role=\\"tab\\" tabindex=\\"-1\\" aria-selected=\\"false\\" aria-controls=\\"panel-6\\" title=\\"Visit Panel 6\\"><span class=\\"slds-assistive-text\\">Panel 6</span></a></li>
        <li class=\\"slds-carousel__indicator slds-m-horizontal_xx-small\\"
          role=\\"presentation\\"><a id=\\"indicator-id-7\\" class=\\"slds-carousel__indicator-action\\" role=\\"tab\\" tabindex=\\"-1\\" aria-selected=\\"false\\" aria-controls=\\"panel-7\\" title=\\"Visit Panel 7\\"><span class=\\"slds-assistive-text\\">Panel 7</span></a></li>
      </ul>
    </div>
  </div>
</div>"
`;

exports[`Carousel Component Default Snapshot - 3 items and navigation 1`] = `
"<div style=\\"max-width:1280px\\">
  <div class=\\"slds-carousel\\" id=\\"ExampleCarouselWith3items\\">
    <div class=\\"slds-grid_vertical slds-col slds-path__scroller\\">
      <div class=\\"slds-grid slds-grid_align-center\\">
        <div class=\\"slds-carousel__stage slds-align_absolute-center\\">
          <div class=\\"slds-carousel__panels slds-is-relative\\" style=\\"transform:translateX(-1000000px)\\">
            <div id=\\"content-id-1\\" class=\\"slds-carousel__panel slds-m-horizontal_xx-small slds-list_horizontal\\" role=\\"tabpanel\\" aria-hidden=\\"false\\" aria-labelledby=\\"indicator-id-1\\" style=\\"max-width:NaNpx\\">
              <a href=\\"javascript:void(0);\\" class=\\"slds-carousel__panel-action slds-text-link_reset\\">
                <div class=\\"slds-carousel__image\\"><img src=\\"assets/images/carousel/carousel-01.jpg\\" alt=\\"Card image description\\" /></div>
                <div class=\\"slds-carousel__content\\">
                  <h2 class=\\"slds-carousel__content-title\\">Carousel Item 1</h2>
                  <p class=\\"slds-p-bottom_x-small slds-text-body_small\\" style=\\"min-height:40px\\">Description for #1</p><a class=\\"slds-button slds-button_neutral slds-button_outline-brand\\" href=\\"https://www.salesforce.com\\" target=\\"_blank\\">Get Started</a></div>
              </a>
            </div>
            <div id=\\"content-id-2\\" class=\\"slds-carousel__panel slds-m-horizontal_xx-small slds-list_horizontal\\" role=\\"tabpanel\\" aria-hidden=\\"false\\" aria-labelledby=\\"indicator-id-2\\" style=\\"max-width:NaNpx\\">
              <a href=\\"javascript:void(0);\\" class=\\"slds-carousel__panel-action slds-text-link_reset\\">
                <div class=\\"slds-carousel__image\\"><img src=\\"assets/images/carousel/carousel-02.jpg\\" alt=\\"Card image description\\" /></div>
                <div class=\\"slds-carousel__content\\">
                  <h2 class=\\"slds-carousel__content-title\\">Carousel Item 2</h2>
                  <p class=\\"slds-p-bottom_x-small slds-text-body_small\\" style=\\"min-height:40px\\">Description for #2 will be much longer than others. It&#x27;ll look ok</p><a class=\\"slds-button slds-button_neutral slds-button_outline-brand\\" href=\\"https://www.salesforce.com\\" target=\\"_blank\\">Get Started</a></div>
              </a>
            </div>
            <div id=\\"content-id-3\\" class=\\"slds-carousel__panel slds-m-horizontal_xx-small slds-list_horizontal\\" role=\\"tabpanel\\" aria-hidden=\\"false\\" aria-labelledby=\\"indicator-id-3\\" style=\\"max-width:NaNpx\\">
              <a href=\\"javascript:void(0);\\" class=\\"slds-carousel__panel-action slds-text-link_reset\\">
                <div class=\\"slds-carousel__image\\"><img src=\\"assets/images/carousel/carousel-03.jpg\\" alt=\\"Card image description\\" /></div>
                <div class=\\"slds-carousel__content\\">
                  <h2 class=\\"slds-carousel__content-title\\">Carousel Item 3</h2>
                  <p class=\\"slds-p-bottom_x-small slds-text-body_small\\" style=\\"min-height:40px\\">Description for #3</p><a class=\\"slds-button slds-button_neutral slds-button_outline-brand\\" href=\\"https://www.salesforce.com\\" target=\\"_blank\\">Get Started</a></div>
              </a>
            </div>
            <div id=\\"content-id-4\\" class=\\"slds-carousel__panel slds-m-horizontal_xx-small slds-list_horizontal\\" role=\\"tabpanel\\" aria-hidden=\\"false\\" aria-labelledby=\\"indicator-id-4\\" style=\\"max-width:NaNpx\\">
              <a href=\\"javascript:void(0);\\" class=\\"slds-carousel__panel-action slds-text-link_reset\\">
                <div class=\\"slds-carousel__image\\"><img src=\\"assets/images/carousel/carousel-02.jpg\\" alt=\\"Card image description\\" /></div>
                <div class=\\"slds-carousel__content\\">
                  <h2 class=\\"slds-carousel__content-title\\">Carousel Item 4</h2>
                  <p class=\\"slds-p-bottom_x-small slds-text-body_small\\" style=\\"min-height:40px\\">Description for #4</p><a class=\\"slds-button slds-button_neutral slds-button_outline-brand\\" href=\\"https://www.salesforce.com\\" target=\\"_blank\\">Get Started</a></div>
              </a>
            </div>
            <div id=\\"content-id-5\\" class=\\"slds-carousel__panel slds-m-horizontal_xx-small slds-list_horizontal\\" role=\\"tabpanel\\" aria-hidden=\\"false\\" aria-labelledby=\\"indicator-id-5\\" style=\\"max-width:NaNpx\\">
              <a href=\\"javascript:void(0);\\" class=\\"slds-carousel__panel-action slds-text-link_reset\\">
                <div class=\\"slds-carousel__image\\"><img src=\\"assets/images/carousel/carousel-01.jpg\\" alt=\\"Card image description\\" /></div>
                <div class=\\"slds-carousel__content\\">
                  <h2 class=\\"slds-carousel__content-title\\">Carousel Item 5</h2>
                  <p class=\\"slds-p-bottom_x-small slds-text-body_small\\" style=\\"min-height:40px\\">Description for #3</p><a class=\\"slds-button slds-button_neutral slds-button_outline-brand\\" href=\\"https://www.salesforce.com\\" target=\\"_blank\\">Learn More</a></div>
              </a>
            </div>
            <div id=\\"content-id-6\\" class=\\"slds-carousel__panel slds-m-horizontal_xx-small slds-list_horizontal\\" role=\\"tabpanel\\" aria-hidden=\\"false\\" aria-labelledby=\\"indicator-id-6\\" style=\\"max-width:NaNpx\\">
              <a href=\\"javascript:void(0);\\" class=\\"slds-carousel__panel-action slds-text-link_reset\\">
                <div class=\\"slds-carousel__image\\"><img src=\\"assets/images/carousel/carousel-03.jpg\\" alt=\\"Card image description\\" /></div>
                <div class=\\"slds-carousel__content\\">
                  <h2 class=\\"slds-carousel__content-title\\">Carousel Item 6</h2>
                  <p class=\\"slds-p-bottom_x-small slds-text-body_small\\" style=\\"min-height:40px\\">Description for #6</p><a class=\\"slds-button slds-button_neutral slds-button_outline-brand\\" href=\\"https://www.salesforce.com\\" target=\\"_blank\\">Learn More</a></div>
              </a>
            </div>
            <div id=\\"content-id-7\\" class=\\"slds-carousel__panel slds-m-horizontal_xx-small slds-list_horizontal\\" role=\\"tabpanel\\" aria-hidden=\\"false\\" aria-labelledby=\\"indicator-id-7\\" style=\\"max-width:NaNpx\\">
              <a href=\\"javascript:void(0);\\" class=\\"slds-carousel__panel-action slds-text-link_reset\\">
                <div class=\\"slds-carousel__image\\"><img src=\\"assets/images/carousel/carousel-02.jpg\\" alt=\\"Card image description\\" /></div>
                <div class=\\"slds-carousel__content\\">
                  <h2 class=\\"slds-carousel__content-title\\">Carousel Item 7</h2>
                  <p class=\\"slds-p-bottom_x-small slds-text-body_small\\" style=\\"min-height:40px\\">Description for #6</p><a class=\\"slds-button slds-button_neutral slds-button_outline-brand\\" href=\\"https://www.salesforce.com\\" target=\\"_blank\\">Learn More</a></div>
              </a>
            </div>
          </div>
        </div>
      </div>
      <ul class=\\"slds-carousel__indicators slds-col slds-text-align_center\\" role=\\"tablist\\">
        <li class=\\"slds-carousel__indicator slds-m-horizontal_xx-small\\" role=\\"presentation\\"><a id=\\"indicator-id-1\\" class=\\"slds-carousel__indicator-action slds-is-active\\" role=\\"tab\\" tabindex=\\"0\\" aria-selected=\\"true\\" aria-controls=\\"panel-1\\" title=\\"Visit Panel 1\\"><span class=\\"slds-assistive-text\\">Panel 1</span></a></li>
        <li class=\\"slds-carousel__indicator slds-m-horizontal_xx-small\\"
          role=\\"presentation\\"><a id=\\"indicator-id-2\\" class=\\"slds-carousel__indicator-action\\" role=\\"tab\\" tabindex=\\"-1\\" aria-selected=\\"false\\" aria-controls=\\"panel-2\\" title=\\"Visit Panel 2\\"><span class=\\"slds-assistive-text\\">Panel 2</span></a></li>
        <li class=\\"slds-carousel__indicator slds-m-horizontal_xx-small\\"
          role=\\"presentation\\"><a id=\\"indicator-id-3\\" class=\\"slds-carousel__indicator-action\\" role=\\"tab\\" tabindex=\\"-1\\" aria-selected=\\"false\\" aria-controls=\\"panel-3\\" title=\\"Visit Panel 3\\"><span class=\\"slds-assistive-text\\">Panel 3</span></a></li>
      </ul>
    </div>
  </div>
</div>"
`;

exports[`Carousel Component With Custom Rendering 1`] = `
"<div class=\\"slds-carousel\\" id=\\"Custom Rendering\\">
  <div class=\\"slds-grid_vertical slds-col slds-path__scroller\\">
    <div class=\\"slds-grid slds-grid_align-center\\">
      <div class=\\"slds-carousel__stage slds-align_absolute-center\\">
        <div class=\\"slds-carousel__panels slds-is-relative\\" style=\\"transform:translateX(-1000000px)\\">
          <div id=\\"content-id-1\\" class=\\"slds-carousel__panel slds-m-horizontal_xx-small slds-list_horizontal\\" role=\\"tabpanel\\" aria-hidden=\\"false\\" aria-labelledby=\\"indicator-id-1\\" style=\\"max-width:NaNpx\\">
            <div class=\\"slds-carousel-custom-rendering\\">
              <div class=\\"slds-carousel__content-title\\">Carousel Item 1 With Custom Rendering</div>
              <div class=\\"slds-carousel__image\\"><img src=\\"assets/images/carousel/carousel-01.jpg\\" alt=\\"Card image description\\" /></div>
            </div>
          </div>
          <div id=\\"content-id-2\\" class=\\"slds-carousel__panel slds-m-horizontal_xx-small slds-list_horizontal\\" role=\\"tabpanel\\" aria-hidden=\\"false\\" aria-labelledby=\\"indicator-id-2\\" style=\\"max-width:NaNpx\\">
            <div class=\\"slds-carousel-custom-rendering\\">
              <div class=\\"slds-carousel__content-title\\">Carousel Item 2 With Custom Rendering</div>
              <div class=\\"slds-carousel__image\\"><img src=\\"assets/images/carousel/carousel-02.jpg\\" alt=\\"Card image description\\" /></div>
            </div>
          </div>
          <div id=\\"content-id-3\\" class=\\"slds-carousel__panel slds-m-horizontal_xx-small slds-list_horizontal\\" role=\\"tabpanel\\" aria-hidden=\\"false\\" aria-labelledby=\\"indicator-id-3\\" style=\\"max-width:NaNpx\\">
            <div class=\\"slds-carousel-custom-rendering\\">
              <div class=\\"slds-carousel__content-title\\">Carousel Item 3 With Custom Rendering</div>
              <div class=\\"slds-carousel__image\\"><img src=\\"assets/images/carousel/carousel-03.jpg\\" alt=\\"Card image description\\" /></div>
            </div>
          </div>
          <div id=\\"content-id-4\\" class=\\"slds-carousel__panel slds-m-horizontal_xx-small slds-list_horizontal\\" role=\\"tabpanel\\" aria-hidden=\\"false\\" aria-labelledby=\\"indicator-id-4\\" style=\\"max-width:NaNpx\\">
            <div class=\\"slds-carousel-custom-rendering\\">
              <div class=\\"slds-carousel__content-title\\">Carousel Item 4 With Custom Rendering</div>
              <div class=\\"slds-carousel__image\\"><img src=\\"assets/images/carousel/carousel-02.jpg\\" alt=\\"Card image description\\" /></div>
            </div>
          </div>
          <div id=\\"content-id-5\\" class=\\"slds-carousel__panel slds-m-horizontal_xx-small slds-list_horizontal\\" role=\\"tabpanel\\" aria-hidden=\\"false\\" aria-labelledby=\\"indicator-id-5\\" style=\\"max-width:NaNpx\\">
            <div class=\\"slds-carousel-custom-rendering\\">
              <div class=\\"slds-carousel__content-title\\">Carousel Item 5 With Custom Rendering</div>
              <div class=\\"slds-carousel__image\\"><img src=\\"assets/images/carousel/carousel-01.jpg\\" alt=\\"Card image description\\" /></div>
            </div>
          </div>
          <div id=\\"content-id-6\\" class=\\"slds-carousel__panel slds-m-horizontal_xx-small slds-list_horizontal\\" role=\\"tabpanel\\" aria-hidden=\\"false\\" aria-labelledby=\\"indicator-id-6\\" style=\\"max-width:NaNpx\\">
            <div class=\\"slds-carousel-custom-rendering\\">
              <div class=\\"slds-carousel__content-title\\">Carousel Item 6 With Custom Rendering</div>
              <div class=\\"slds-carousel__image\\"><img src=\\"assets/images/carousel/carousel-03.jpg\\" alt=\\"Card image description\\" /></div>
            </div>
          </div>
          <div id=\\"content-id-7\\" class=\\"slds-carousel__panel slds-m-horizontal_xx-small slds-list_horizontal\\" role=\\"tabpanel\\" aria-hidden=\\"false\\" aria-labelledby=\\"indicator-id-7\\" style=\\"max-width:NaNpx\\">
            <div class=\\"slds-carousel-custom-rendering\\">
              <div class=\\"slds-carousel__content-title\\">Carousel Item 7 With Custom Rendering</div>
              <div class=\\"slds-carousel__image\\"><img src=\\"assets/images/carousel/carousel-02.jpg\\" alt=\\"Card image description\\" /></div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <ul class=\\"slds-carousel__indicators slds-col slds-text-align_center\\" role=\\"tablist\\">
      <li class=\\"slds-carousel__indicator slds-m-horizontal_xx-small\\" role=\\"presentation\\"><a id=\\"indicator-id-1\\" class=\\"slds-carousel__indicator-action slds-is-active\\" role=\\"tab\\" tabindex=\\"0\\" aria-selected=\\"true\\" aria-controls=\\"panel-1\\" title=\\"Visit Panel 1\\"><span class=\\"slds-assistive-text\\">Panel 1</span></a></li>
      <li class=\\"slds-carousel__indicator slds-m-horizontal_xx-small\\"
        role=\\"presentation\\"><a id=\\"indicator-id-2\\" class=\\"slds-carousel__indicator-action\\" role=\\"tab\\" tabindex=\\"-1\\" aria-selected=\\"false\\" aria-controls=\\"panel-2\\" title=\\"Visit Panel 2\\"><span class=\\"slds-assistive-text\\">Panel 2</span></a></li>
      <li class=\\"slds-carousel__indicator slds-m-horizontal_xx-small\\"
        role=\\"presentation\\"><a id=\\"indicator-id-3\\" class=\\"slds-carousel__indicator-action\\" role=\\"tab\\" tabindex=\\"-1\\" aria-selected=\\"false\\" aria-controls=\\"panel-3\\" title=\\"Visit Panel 3\\"><span class=\\"slds-assistive-text\\">Panel 3</span></a></li>
      <li class=\\"slds-carousel__indicator slds-m-horizontal_xx-small\\"
        role=\\"presentation\\"><a id=\\"indicator-id-4\\" class=\\"slds-carousel__indicator-action\\" role=\\"tab\\" tabindex=\\"-1\\" aria-selected=\\"false\\" aria-controls=\\"panel-4\\" title=\\"Visit Panel 4\\"><span class=\\"slds-assistive-text\\">Panel 4</span></a></li>
    </ul>
  </div>
</div>"
`;
